packages/astro/src/internal/swap-document.ts

```diff
@@ -0,0 +1,61 @@
+// eslint-disable-next-line import/no-unresolved
+import { swapFunctions } from 'astro:transitions/client';
+
+const PERSIST_ATTR = 'data-astro-transition-persist';
+const EMOTION_ATTR = 'data-emotion';
+
+/**
+ * @internal
+ * Custom swap function to make mounting and styling
+ * of Clerk components work with View Transitions in Astro.
+ *
+ * See https://docs.astro.build/en/guides/view-transitions/#building-a-custom-swap-function
+ */
+export function swapDocument(doc: Document) {
+  swapFunctions.deselectScripts(doc);
+  swapFunctions.swapRootAttributes(doc);
+
+  // Keep the elements created by `@emotion/cache`
+  const emotionElements = document.querySelectorAll(`style[${EMOTION_ATTR}]`);
+  swapHeadElements(doc, Array.from(emotionElements));
```

Comment:
We need to keep the `<style>` element added by `@emotion/cache` to keep the styling when switching between pages.

<img width="363" alt="Screenshot 2024-10-17 at 1 34 41 PM" src="https://github.com/user-attachments/assets/ce9be2d1-5608-4b31-9cc7-c27f3ab80f2d">

The default Astro swap function will remove this one